CMakeLists.txt

```diff
@@ -16,32 +13,58 @@ if(NOT CMAKE_BUILD_TYPE)
    set(CMAKE_BUILD_TYPE Release)
 endif()
 
-option(SMF_INSTALL "Enable installation of smf" ON)
-option(SMF_ENABLE_TESTS "Useful for disabling all tests" ON)
+option(SMF_BUILD_PROGRAMS "Build smf programs and demos" ON)
 option(SMF_ENABLE_INTEGRATION_TESTS "control if integrations are bulit and ran" ON)
 option(SMF_ENABLE_UNIT_TESTS "control if unit tests are bulit and ran" ON)
 option(SMF_ENABLE_BENCHMARK_TESTS "control if benchmarks are bulit and ran" OFF)
-option(SMF_ENABLE_CMAKE_PROJECT_FLAGS "control cmake_cxx_flags_*" OFF)
-option(SMF_BUILD_PROGRAMS "Build smf programs and demos" ON)
+option(SMF_INSTALL "Enable installation of smf" ON)
+option(SMF_ENABLE_TESTS "Useful for disabling all tests" ON)
 
-if(NOT SMF_ENABLE_TESTS)
-  set(SMF_ENABLE_INTEGRATION_TESTS  OFF)
-  set(SMF_ENABLE_UNIT_TESTS  OFF)
-  set(SMF_ENABLE_BENCHMARK_TESTS  OFF)
+# fetch and build dependencies. each dependency is built in isolation allowing
+# tight control over the build configuration, and avoiding any inadvertant cmake
+# cache pollution. the dependencies are built at configure-time and then
+# imported as normal cmake dependencies, which will easily enable future
+# iterations to selectively satisfy dependencies from system packages.
+configure_file(CMakeLists.txt.in deps-build/CMakeLists.txt @ONLY)
```

Comment:
why not in cmake_current_binary_dir/cmakelist.txt

Reply:
i think it should be in an isolated directory, but yes, `cmake_current_binary_dir/deps-build/cmakelists.txt`. is there a reason to not do that? my reasoning is that the incapsulating cmake project already _owns_ that directory.

Reply:
yeah we are on the same page w.r.t `cmake_current_binary_dir/deps-build/cmakelists.txt`